phase1-coordinator/src/coordinator.rs

```diff
@@ -2130,8 +2230,8 @@ impl Coordinator {
 
         // Check the justification and extract the tasks.
         let (tasks, replacement) = match justification {
-            Justification::BanCurrent(_, _, _, ref tasks, ref replacement) => (tasks, replacement),
-            Justification::DropCurrent(_, _, _, ref tasks, ref replacement) => (tasks, replacement),
+            Justification::BanCurrent(data) => (&data.tasks, &data.replacement),
+            Justification::DropCurrent(data) => (&data.tasks, &data.replacement),
```

Comment:
Using of a struct with named fields instead of a tuple struct is so much better!